citrix_hypervisor/datadog_checks/citrix_hypervisor/check.py

```diff
@@ -0,0 +1,182 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Any, Dict, List
+
+import yaml
+from six.moves import xmlrpc_client as xmlrpclib
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+from .metrics import build_metric
+
+
+class CitrixHypervisorCheck(AgentCheck):
+    __NAMESPACE__ = 'citrix_hypervisor'
+    SERVICE_CHECK_CONNECT = 'can_connect'
+
+    def __init__(self, name, init_config, instances):
+        # type: (str, Dict, List[Dict]) -> None
+        super(CitrixHypervisorCheck, self).__init__(name, init_config, instances)
+
+        self._last_timestamp = 0
```

Comment:
when `self._last_timestamp = 0`, does it return metrics from beginning of time or is there a cap? 

If the customer restarts the agent or upgrades the agent, would the check start submitting metrics from the past, resulting in duplicate data or data submitted under incorrect time assumptions?

Reply:
If the timestamp argument in the http request is `start=0`, the endpoint returns metrics from the beginning, with a granularity of 3600s. Note that at the initialization, we update the latest timestamp https://github.com/DataDog/integrations-core/pull/9815/files#diff-9f503b54514923c0d47f4196654337826d9416f30255afd407866a5bc8828d6aR39

Since we're only parsing the last row of metrics returned, if the agent restarts and the timestamp initialization fails, we might get duplicated data for one point in time, but I think that's fine